tls/s2n_client_hello.c

```diff
@@ -257,6 +257,18 @@ static int s2n_parse_client_hello(struct s2n_connection *conn)
     return S2N_SUCCESS;
 }
 
+bool s2n_is_tls_12_self_downgrade_required(struct s2n_connection *conn) {
+    /* RSA PSS is required for TLS 1.3 connections. So if there's a possibility that an RSA Certificate could be picked
+     * by a client connection, then downgrade connection to TLS 1.2 if our libcrypto doesn't support RSA PSS. */
+    if ((conn->mode == S2N_SERVER)
```

Comment:
Is this mode check required? Would it ever make sense for the client to call this method?

Reply:
A client should never call this method since the server gets to decide the TLS Version. The client should advertise it's highest supported TLS Version, all of it's supported signature algorithms, and let the server decide the TLS Version.

In order to avoid relying only on good intentions, I added this `(conn->mode == S2N_SERVER)` check to ensure that even if this function is ever erroneously called client-side, that clients still would not self-downgrade.